jkube-kit/generator/api/src/main/java/org/eclipse/jkube/generator/api/support/BaseGenerator.java

```diff
@@ -204,6 +205,9 @@ protected String getAlias() {
     }
 
     protected boolean shouldAddImageConfiguration(List<ImageConfiguration> configs) {
+        if (isSimpleDockerfileMode()) {
+            return false;
+        }
         return !containsBuildConfiguration(configs) || Configs.asBoolean(getConfig(Config.add));
```

Comment:
It would be good to split this statement too:
```suggestion
        if (containsBuildConfiguration(configs)) {
            return false;
        }        
        return Configs.asBoolean(getConfig(Config.add));
```

Reply:
Trying to figure out the logic within this [statement](https://www.eclipse.org/jkube/docs/kubernetes-maven-plugin#generators-default):

> add: When this set to true, then the generator adds to an existing image configuration. By default this is disabled, so that a generator only kicks in when there are no other image configurations in the build, which are either configured directly for a k8s:build or already added by a generator which has been run previously.

I **think** this should translate into this:
```
        if (containsBuildConfiguration(configs)) {
            return Configs.asBoolean(getConfig(Config.add));
        }        
        return true;
```

Which is more or less what was there and what you did, but the logic is more readable (IMHO)

I would also change the method name to `shouldAddGeneratedImageConfiguration`